src/azure-cli/azure/cli/command_modules/profile/tests/latest/test_profile_custom.py

```diff
@@ -124,6 +124,21 @@ def test_login(msi_port, identity_id=None):
         # assert
         self.assertTrue(invoked)
 
+    @mock.patch('azure.cli.core._profile.SubscriptionFinder', autospec=True)
+    def test_login_invalid_url(self, mock_subscription_finder):
+        from requests.exceptions import InvalidURL
+        from azure.cli.core.azclierror import UnclassifiedUserFault
+
+        class SubscriptionFinderStub:
+            def find_through_authorization_code_flow(self, tenant, resource, authority_url):
+                raise InvalidURL(request='http://some.unknown.endpoints')
```

Comment:
A browse won't be available on CI machine, so **auth code flow** won't be used.

Reply:
Appreciate for your help! That makes sense why CI fails....